refiner-lib/src/refiner_inner.rs

```diff
@@ -432,6 +439,42 @@ fn build_transaction(
                         txs.get(&hash),
                     )?
                 }
+                InnerTransactionKind::Deploy | InnerTransactionKind::DeployErc20 => {
+                    hash = virtual_receipt_id.0.try_into().unwrap();
+                    let eth_tx: NormalizedEthTransaction =
```

Comment:
Here I had to pull the `eth_tx` to get the `nonce`. Not sure if this is ok.

Reply:
This doesn't work unfortunately. The payload for `deploy_code` is raw EVM bytecode ([see Engine implementation](https://github.com/aurora-is-near/aurora-engine/blob/master/engine/src/lib.rs#L265)), not an EVM transaction (and this makes sense because `deploy_code` is meant to be a NEAR-native way of interacting with the EVM, not a metatransaction like `submit`). To get the right address in this case we'll need to make a function call to the built-in Borealis Engine instance. I suggest we do this in a separate PR though because I suspect almost no one cares about us indexing `deploy` correctly since it gets almost no traffic. So better to not hold up this PR with complexity that isn't meeting much of a need in my opinion.

The `deploy_erc20_token` case on the other hand is easy and  think we should handle it here. The output from that method is simply the address bytes. So if the transaction was successful then we can get fill the `contract_address` field from the output directly, and if the transaction was not successful then we skip indexing it anyway (I think failed NEAR transactions are ignored before we get to this point in the logic). The only reason we needed to introduce the `create_legacy_address` function in the `submit` case is because we need to report the `contract_address` even when we get back `TransactionStatus::Revert` (and in that case the would-be created address is not present in the output).